presto-hive/src/test/java/com/facebook/presto/hive/s3/security/TestAWSS3SecurityMapping.java

```diff
@@ -0,0 +1,198 @@
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+public class TestAWSS3SecurityMapping
+{
+    private static final String DEFAULT_USER = "defaultUser";
+
+    @Test
+    public void testAWSS3SecurityMapping()
+    {
+        AWSSecurityMappingConfig mappingConfig = new AWSSecurityMappingConfig()
+                .setConfigFile(new File(getResource(getClass(), "aws-security-mapping-with-default-role.json").getPath()));
+
+        AWSSecurityMappingsSupplier awsSecurityMappingsSupplier =
+                new AWSSecurityMappingsSupplier(mappingConfig.getConfigFile(), mappingConfig.getRefreshPeriod());
+
+        DynamicConfigurationProvider provider = new AWSS3SecurityMappingConfigurationProvider(awsSecurityMappingsSupplier);
+
+        // matches user -- mapping provides credentials
+        assertMapping(
+                provider,
+                empty().withUser("admin"),
+                credentials("AKIAxxxaccess", "iXbXxxxsecret"));
+
+        // matches user regex -- mapping provides iam role
+        assertMapping(
+                provider,
+                empty().withUser("analyst"),
+                role("arn:aws:iam::123456789101:role/analyst_and_scientist_role"));
+
+        // matches empty rule at end -- default role used
+        assertMapping(
+                provider,
+                empty().withUser("defaultUser"),
+                role("arn:aws:iam::123456789101:role/default"));
+    }
+
+    @Test
+    public void testFailAWSS3SecurityMapping()
+    {
+        AWSSecurityMappingConfig mappingConfig = new AWSSecurityMappingConfig()
+                .setConfigFile(new File(getResource(getClass(), "aws-security-mapping-without-default-role.json").getPath()));
+
+        AWSSecurityMappingsSupplier awsSecurityMappingsSupplier =
+                new AWSSecurityMappingsSupplier(mappingConfig.getConfigFile(), mappingConfig.getRefreshPeriod());
+
+        DynamicConfigurationProvider provider = new AWSS3SecurityMappingConfigurationProvider(awsSecurityMappingsSupplier);
+
+        // matches no security mapping -- access denied
+        assertMappingFails(
+                provider,
+                empty().withUser("defaultUser"),
+                "No matching AWS S3 security mapping");
+    }
+
+    private static void assertMapping(DynamicConfigurationProvider provider, MappingSelector selector, MappingResult mappingResult)
+    {
+        Configuration configuration = new Configuration(false);
+
+        assertNull(configuration.get(S3_ACCESS_KEY));
+        assertNull(configuration.get(S3_SECRET_KEY));
+        assertNull(configuration.get(S3_IAM_ROLE));
+
+        applyMapping(provider, selector, configuration);
+
+        assertEquals(configuration.get(S3_ACCESS_KEY), mappingResult.getAccessKey().orElse(null));
```

Comment:
orElse in an assert feels strange to me. Just use get(). If it's not present the assert will fail

Reply:
Modified this